### ISSUE TITLE
Numba rescaleData needs np dtypes

### ISSUE THREAD

**Issue by Neywiny** [pyqtgraph/pyqtgraph#1918]
### Short description
Have boolean array, attempting to plot. Enabling useNumba throws errors when converting to ARGB.

### Code to reproduce
```python
from PyQt5 import QtWidgets
from pyqtgraph import PlotWidget
import pyqtgraph as pg
import sys  # We need sys so that we can pass argv to QApplication
import numpy as np
pg.setConfigOption('useNumba', True)
class MainWindow(QtWidgets.QMainWindow):
	def __init__(self, *args, **kwargs):
		super(MainWindow, self).__init__(*args, **kwargs)
		self.graphWidget = pg.PlotWidget()
		self.setCentralWidget(self.graphWidget)
		img = pg.ImageItem(np.random.random((2,2)).astype(bool))
		cmap = pg.ColorMap((0, 1), ((255, 0, 0), (0, 255, 0)))
		lut = cmap.getLookupTable(0, 1, 256)
		img.setLookupTable(lut)
		self.graphWidget.addItem(img)
		self.img = img
app = QtWidgets.QApplication(sys.argv)
main = MainWindow()
main.show()
sys.exit(app.exec_())
```

### Expected behavior
This app should show a random grid of data.

### Real behavior
Does not render(), throws an error instead. In fact, multiple of same error.

```
Traceback (most recent call last):
  File "C:\Users\User\Python38\lib\site-packages\pyqtgraph\graphicsItems\ImageItem.py", line 834, in paint
    self.render()
  File "C:\Users\User\Python38\lib\site-packages\pyqtgraph\graphicsItems\ImageItem.py", line 561, in render
    fn.makeARGB(image, lut=lut, levels=levels, output=self._processingBuffer)
  File "C:\Users\User\Python38\lib\site-packages\pyqtgraph\functions.py", line 1460, in makeARGB
    data = rescaleData(data, scale/rng, minVal, dtype=dtype)
  File "C:\Users\User\Python38\lib\site-packages\pyqtgraph\functions.py", line 1304, in rescaleData
    return numba_fn.rescaleData(data, scale, offset, out_dtype, clip)
  File "C:\Users\User\Python38\lib\site-packages\pyqtgraph\functions_numba.py", line 16, in rescaleData
    func = numba.guvectorize(
  File "C:\Users\User\Python38\lib\site-packages\numba\np\ufunc\decorators.py", line 194, in wrap
    guvec.add(fty)
  File "C:\Users\User\Python38\lib\site-packages\numba\np\ufunc\gufunc.py", line 60, in add
    self.gufunc_builder.add(fty)
  File "C:\Users\User\Python38\lib\site-packages\numba\np\ufunc\ufuncbuilder.py", line 212, in add
    cres, args, return_type = _compile_element_wise_function(
  File "C:\Users\User\Python38\lib\site-packages\numba\np\ufunc\ufuncbuilder.py", line 144, in _compile_element_wise_function
    cres = nb_func.compile(sig, **targetoptions)
  File "C:\Users\User\Python38\lib\site-packages\numba\np\ufunc\ufuncbuilder.py", line 93, in compile
    return self._compile_core(sig, flags, locals)
  File "C:\Users\User\Python38\lib\site-packages\numba\np\ufunc\ufuncbuilder.py", line 124, in _compile_core
    args, return_type = sigutils.normalize_signature(sig)
  File "C:\Users\User\Python38\lib\site-packages\numba\core\sigutils.py", line 29, in normalize_signature
    parsed = _parse_signature_string(sig)
  File "C:\Users\User\Python38\lib\site-packages\numba\core\sigutils.py", line 19, in _parse_signature_string
    return eval(signature_str, {}, types.__dict__)
  File "<string>", line 1, in <module>
TypeError: 'type' object is not subscriptable
```

### Tested environment(s)

 * PyQtGraph version: 0.12.2
 * Qt Python binding: 'PyQt5 5.15.4 Qt 5.15.2'
 * Python version: 3.8.3
 * NumPy version: 1.21.0
 * Operating system: Win 10
 * Installation method: pip

### Additional context
When I drill down into that stacktrace, it is attempting to vectorize "bool[:],f8,f8,f8,f8,uint8[:]" which isn't working. In my code the array is of bools, but it can be switched. Just thought it was an interesting error. Not too sure best fix, might just be to cast. That is how I did it for my code.

**Comment by pijyoi** [comment]
Hi, thank you for reporting. I simplified your example a bit.
The issue is that within ```functions_numba.py::rescaleData()```, we are using ```dtype.name``` for which:
```python
In [2]: np.dtype(bool).name
Out[2]: 'bool'
``` 
However, this does not match Numba's idea of a boolean's name. (Numba expects ```boolean```)
https://numba.pydata.org/numba-doc/dev/reference/types.html

```python
import pyqtgraph as pg
import numpy as np

pg.setConfigOption('useNumba', True)

pg.mkQApp()
pw = pg.PlotWidget()
pw.show()
data = np.random.random((100,100)) > 0.5
img = pg.ImageItem(data)
cmap = pg.ColorMap((0, 1), ((255, 0, 0), (0, 255, 0)))
lut = cmap.getLookupTable(0, 1, 256)
img.setLookupTable(lut)
pw.addItem(img)

pg.exec()
```

**Comment by pijyoi** [comment]
If you are interested in performance, I would suggest the following formulation:
1) use row-major
2) pass in data as uint8 which will bypass makeARGB altogether
    * the rendered image will be of format Indexed8 (1-byte per pixel) rather than ARGB32 (4-bytes per pixel)
3) disable auto-levels since values are known to be 0 or 1

```python
import pyqtgraph as pg
import numpy as np

pg.setConfigOption('imageAxisOrder', 'row-major')

pg.mkQApp()
pw = pg.PlotWidget()
pw.show()
data = np.random.random((100,100)) > 0.5
img = pg.ImageItem(data.view(np.uint8), levels=[0, 1])
cmap = pg.ColorMap((0, 1), ((255, 0, 0), (0, 255, 0)))
lut = cmap.getLookupTable(0, 1, 256)
img.setLookupTable(lut)
pw.addItem(img)

pg.exec()
```

**Comment by Neywiny** [comment]
Hi, thank you. I am new to this package as of a few days ago and haven't done much, so I appreciate the direction. 

1. Thank you. I had to redo a bit of the matrix, but it does look a bit smoother on updates.
2. The array was generated using a < with 2 np vectors. I have now casted it to a u8 to avoid the error. 
3. I thought it might so I tried it, and after my fix I did indeed not notice a huge improvement. But I found this bug.
4. I forgot to do that in the example, it was done in the original code. Thank you for noticing.
